src/devdb/components/qaqc_version_history_report.py

```diff
@@ -0,0 +1,58 @@
+import streamlit as st
+import pandas as pd
+import numpy as np
+import plotly.express as px
+import plotly.graph_objects as go
+from src.constants import COLOR_SCHEME
+import pdb
+
+
+class QAQCVersionHistoryReport:
+    def __init__(self, data, qaqc_check_dict, qaqc_check_groups) -> None:
+        self.qaqc_version_history = data["qaqc_historic"]
+        self.qaqc_checks = qaqc_check_dict
+        self.qaqc_check_groups = qaqc_check_groups
+
+    def __call__(self):
+        st.subheader("Version History for QAQC Checks")
+        for group_name, group_description in self.qaqc_check_groups.items():
+            st.markdown(
+                f"""
+                ### {group_name} Group
+                {group_description}
+                """
+            )
+            self.display_check_distribution(group_name)
+
+    def display_check_distribution(self, group):
+        checks = self.checks_by_group(group)
+        df = self.filter_by_checks(checks)
+        versions = ["21Q4", "21Q2", "20Q4"]
```

Comment:
This could be generated another way - maybe by the user selecting which versions to compare?

Reply:
Yes - this will be improved - wanted to get an eye on the graphs before pushing forward in a direction